vllm/worker/model_runner.py

```diff
@@ -1354,6 +1356,9 @@ def execute_model(
             "finished_requests_ids": model_input.finished_requests_ids,
             "request_ids_to_seq_ids": model_input.request_ids_to_seq_ids,
         } if self.has_seqlen_agnostic else {}
+        model_forward_start = torch.cuda.Event(enable_timing=True)
```

Comment:
what's the overhead of this? should we make it opt-in?

Reply:
the Event creation and recording itself is in the noise. But the synchronize further could be theoretically costly for some combo of model architecture + request type.

Hence that part is protected by a flag. All the other ones - scheduling time, model_execute call are not protected since they are low overhead.